core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -449,6 +458,13 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
    */
   protected def doKillExecutors(executorIds: Seq[String]): Boolean = false
 
+  /**
+   * Executor ids which will possibly be preempted by cluster manager.
+   * @return A set of executor IDs
+   */
+  protected def preemptedExecutors: Set[String] = synchronized {
+    preemptedExecutorIDs.toSet
```

Comment:
Why not store it as a `Set` in the first place? It's not like you're modifying it.